data/perform-sql-updates.sh

```diff
@@ -38,10 +38,41 @@ echo "done."
 
 # apply updates in parallel across tables
 echo -e "\nApplying updates in parallel across tables..."
-psql $PSQLOPTS  $@ -f apply-updates-non-planet-tables.sql &
-psql $PSQLOPTS  $@ -f apply-planet_osm_polygon.sql &
-psql $PSQLOPTS  $@ -f apply-planet_osm_line.sql &
-psql $PSQLOPTS  $@ -f apply-planet_osm_point.sql &
+psql $PSQLOPTS  $@ -e -f apply-updates-non-planet-tables.sql &
+
+# use postgres' own estimate of the percentile breakdown of the osm_id column to
+# guide the distribution of jobs, so hopefully they end up mostly evenly sized.
+for tbl in polygon line point; do
+    sql_script="apply-planet_osm_${tbl}.sql"
+    for pct in 25 50 75; do
```

Comment:
Is the machine we run this on always 4 cores?

Reply:
Ideally, it would be related to the number of cores on the PostgreSQL server, not the machine we're running the script on. However, there's a couple of things that make matching the number of statements to the range of queries hard:

1. Doing this in shell script is _possible_, but makes me uncomfortable. I find that getting shell code correct is a lot harder than it looks. Going to fixed concurrency seemed like a good compromise. We can go to variable concurrency, but I'd want to switch all of this into Python, or something else that has less magic than shell.
2. I'm not aware of any call in PostgreSQL to tell us how many cores the server has, which is the important number to match. We could pass that in as an argument, but again; complexity.

Overall, it seemed to me like a good compromise to get something working quickly. If we want it to be more readable and more robust, then I can take a look at doing that next week?

Reply:
I'm fine with hardcoding it now... just as long as we're getting good use out of the PostggresSQL Server we generally run this on. If that's 4 cores, so be it! :)

Reply:
In the comment, please also note this is configured for a 4 core PostgreSQL machine and would need to be adopted to other cores?

Reply:
Good idea. Fixed in dbebede.